lib/libzfs/libzfs_sendrecv.c

```diff
@@ -3122,6 +3122,11 @@ zfs_receive_one(libzfs_handle_t *hdl, int infd, const char *tosnap,
 			VERIFY(0 == nvlist_add_uint64(props,
 			    zfs_prop_to_name(ZFS_PROP_CANMOUNT), 0));
 		}
+		if (0 == nvlist_lookup_nvlist(fs, "snapprops",
+			&snapprops_nvlist)) {
```

Comment:
It would make sense to introduce a local variable in this block simply to be used to lookup the snapprops nvlist.  As was done with `props` in the original version of this function.  That way `snapprops_nvlist` will only be non-NULL when snapname exists.